django_mongodb_backend/cache.py

```diff
@@ -84,19 +126,30 @@ def get_many(self, keys, version=None):
         with self.collection_for_read.find(
             {"key": {"$in": tuple(keys_map)}, **self._filter_expired(expired=False)}
         ) as cursor:
-            return {keys_map[row["key"]]: self.serializer.loads(row["value"]) for row in cursor}
+            results = {}
+            for row in cursor:
+                try:
+                    results[keys_map[row["key"]]] = self.serializer.loads(row["value"], row["pickled"], row["signature"])
+                except SuspiciousOperation as e:
+                    self.delete(row["key"])
+                    e.add_note(f"Cache entry with key '{row['key']}' was deleted due to suspicious data")
+                    raise e
+            return results
 
     def set(self, key, value, timeout=DEFAULT_TIMEOUT, version=None):
         key = self.make_and_validate_key(key, version=version)
         num = self.collection_for_write.count_documents({}, hint="_id_")
         if num >= self._max_entries:
             self._cull(num)
+        value, pickled, signature = self.serializer.dumps(value)
         self.collection_for_write.update_one(
             {"key": key},
             {
                 "$set": {
                     "key": key,
-                    "value": self.serializer.dumps(value),
+                    "value": value,
+                    "pickled": pickled,
+                    "signature": signature,
```

Comment:
I think it's not good to add "pickled" and "signature" keys to all cache data when signing is disabled.